File: web/react_app/src/repository/Repository.js
require("es6-promise").polyfill(); // 安卓4.3以下不兼容 promise, 无法使用 axios, 该库可以解决这个问题, 必须在使用 axios 之前执行
const axios = require("axios");

/**
 * 所有的网络请求
 * 所有的本地存储
 */
class Repository {
    /**
     * 登录
     * @param requestData
     * @param onSuccess
     * @param onFailure
     */
    static requestLogin(requestData, onSuccess, onFailure) {
        this.request("/login", requestData, onSuccess, onFailure);
    }

    static request(url, requestData, onSuccess, onFailure) {
        console.warn("request data:" + requestData);
        axios
            .post(url, requestData)
            .then(function(response) {
                console.warn("response data:" + response);
                if (response && response.code === 200) {
                    onSuccess(response.data);
                } else {
                    throw Error("response code invalid");
                }
            })
            .catch(function(error) {
                console.warn("response error:" + error);
                onFailure(error);
            });
    }
}

export default Repository;

File: mobile/flutter/lib_flutter_base/example/lib/repository/repository.dart
import 'dart:async';
import 'dart:io';

import 'package:dio/dio.dart';

//import 'package:smart/com/smart/library/base/utils/stbase_http_manager.dart';
import 'response_model.dart';

class Repository {
  /// base request
  static Future<ResponseModel> _requestFormPost(String path,
      {data, Options options, CancelToken cancelToken}) async {
    Completer<ResponseModel> completer = new Completer();
    /* HttpManager.getClient().post(path, data: data, options: options, cancelToken: cancelToken).then((response) {
      var responseData = response.data;
      var responseModel = ResponseModel(responseData["code"], responseData["msg"], responseData["time"], responseData["data"]);
      if (responseModel.code == 1) {
        completer.complete(responseModel);
      } else {
        completer.completeError("${responseModel.msg}");
      }
    }).catchError((error) => completer.completeError(error));*/

    return completer.future;
  }

  // -------------------------------------------------------------------------------------------------------------------------
  // business request
  // -------------------------------------------------------------------------------------------------------------------------

  static Future<dynamic> requestLogin(String mobile, String password) async {
    Completer<dynamic> completer = new Completer();

    _requestFormPost("/user/login",
            data: new FormData.fromMap({"mobile": mobile, "password": password}))
        .then((ResponseModel responseModel) {
      completer.complete(responseModel.data);
    }).catchError((error) {
      completer.completeError(error);
    });

    return completer.future;
  }

  static Future<dynamic> requestAuth(int id, String surname, String name,
      String idcard, File image1, File image2, File image3) async {
    Completer<dynamic> completer = new Completer();
    _requestFormPost("/user/auth_man",
        data: new FormData.fromMap({
          "id": id,
          "surname": surname,
          "name": name,
          "image1":  MultipartFile.fromFileSync(image1.path, filename:"image1"),
          "image2": MultipartFile.fromFileSync(image2.path, filename:"image2"),
          "image3": MultipartFile.fromFileSync(image3.path, filename:"image3"),
          "idcard": idcard,
        })).then((ResponseModel responseModel) {
      completer.complete(responseModel.data);
    }).catchError((error) {
      completer.completeError(error);
    });

    return completer.future;
  }

  static Future<dynamic> requestMyDeal(int id) async {
    Completer<dynamic> completer = new Completer();
    _requestFormPost("/user/my_deal", data: new FormData.fromMap({"id": id}))
        .then((ResponseModel responseModel) {
      completer.complete(responseModel.data);
    }).catchError((error) {
      completer.completeError(error);
    });

    return completer.future;
  }
}
